mkosi/machine.py

```diff
@@ -0,0 +1,97 @@
+# SPDX-License-Identifier: LGPL-2.1+
+
+from __future__ import annotations
+
+import contextlib
+import signal
+import sys
+from typing import Any, List, Optional
+
+import pexpect  # type: ignore
+
+import mkosi
+
+from .backend import die
+
+
+class Machine():
+    def __init__(self, args: Optional[List[str]] = None) -> None:
```

Comment:
Let's make args `Sequence[str]` instead of `List[str]`. `Sequence` is a read-only iterable and since we're not gonna modify it we should specify so instead of fixing the type to be `List`.

Reply:
This should work now that #898 landed